assets/src/pages/room/VideochatSection.tsx

```diff
@@ -49,66 +85,93 @@ const prepareScreenSharingStreams = (
-        peerName: peerName,
+        displayName: peerName ?? "Unknown",
       })
     );
 
-  const screenSharingStreams: VideoStreamWithMetadata[] = localPeer?.tracks["screensharing"]?.stream
+  const screenSharingStreams: ScreenShareTileConfig[] = localPeer?.tracks["screensharing"]?.stream
     ? [localPeerToScreenSharingStream(localPeer), ...peersScreenSharingTracks]
     : peersScreenSharingTracks;
 
-  const isScreenSharingActive: boolean = screenSharingStreams.length > 0;
-  return { screenSharingStreams, isScreenSharingActive };
+  return screenSharingStreams;
 };
 
 const remoteTrackToLocalTrack = (localPeer: Track | undefined): TrackWithId | null =>
   localPeer ? { ...localPeer, remoteTrackId: localPeer.trackId } : null;
 
+const takeOutPinnedTiles = (
+  tiles: MediaPlayerTileConfig[],
+  pinnedTileIds: string[]
+): { pinnedTiles: MediaPlayerTileConfig[]; unpinnedTiles: MediaPlayerTileConfig[] } => {
+  const { pinnedTiles, unpinnedTiles } = groupBy(tiles, ({ mediaPlayerId }) =>
+    pinnedTileIds.includes(mediaPlayerId) ? "pinnedTiles" : "unpinnedTiles"
+  );
+  return { pinnedTiles: pinnedTiles ?? [], unpinnedTiles: unpinnedTiles ?? [] };
+};
+
 export const VideochatSection: FC<Props> = ({ peers, localPeer, showSimulcast, webrtc }: Props) => {
   const video: TrackWithId | null = remoteTrackToLocalTrack(localPeer?.tracks["camera"]);
   const audio: TrackWithId | null = remoteTrackToLocalTrack(localPeer?.tracks["audio"]);
 
-  const localUser: MediaPlayerTileConfig = {
-    peerId: localPeer?.id,
+  const localUser: LocalTileConfig = {
+    typeName: "local",
+    peerId: localPeer?.id ?? "Unknown",
     displayName: LOCAL_PEER_NAME,
     initials: computeInitials(localPeer?.metadata?.displayName || ""),
-    video: video ? [video] : [],
-    audio: audio ? [audio] : [],
-    flipHorizontally: true,
+    video: video,
+    audio: audio,
     streamSource: "local",
-    playAudio: false,
     mediaPlayerId: LOCAL_VIDEO_ID,
   };
 
-  const { screenSharingStreams, isScreenSharingActive } = prepareScreenSharingStreams(peers, localPeer);
-  const noPeers = !peers.length;
+  const screenSharingStreams = prepareScreenSharingStreams(peers, localPeer);
+
+  const allPeersConfig: MediaPlayerTileConfig[] = [localUser, ...mapRemotePeersToMediaPlayerConfig(peers)];
+  const allTilesConfig: MediaPlayerTileConfig[] = allPeersConfig.concat(screenSharingStreams);
+
+  const pinningApi = usePinning();
+  const { pinnedTiles, unpinnedTiles } = takeOutPinnedTiles(allTilesConfig, pinningApi.pinnedTileIds);
+  const isSomeTilePinned = pinnedTiles.length > 0;
+
+  const wrapperClass = useMemo(() => {
+    const base = "grid h-full w-full auto-rows-fr gap-3 3xl:max-w-[1728px]";
+    const layoutWithTileHighlight = allTilesConfig.length === 2 ? "relative" : "sm:grid-cols-3/1";
+
+    return clsx(base, isSomeTilePinned && layoutWithTileHighlight);
+  }, [isSomeTilePinned, allTilesConfig.length]);
 
+  const shouldBlockPinning = unpinnedTiles.length === 1;
```

Comment:
How hard would it be to handle that scenario too? I can imagine you'd like to pin all the items before someone joins, e.g. you're pinning screen share and your camera before the other peer arrives. At first I thought it was a bug that the button didn't appear

Reply:
Thank you. I'll talk to Alicja about those issues.